mathics/builtin/atomic/symbols.py

```diff
@@ -59,36 +59,134 @@ def _get_usage_string(symbol, evaluation, is_long_form: bool):
     definition = evaluation.definitions.get_definition(symbol.name)
     ruleusage = definition.get_values_list("messages")
     usagetext = None
-    import re
 
     # First look at user definitions:
     for rulemsg in ruleusage:
         if rulemsg.pattern.expr.elements[1].__str__() == '"usage"':
             usagetext = rulemsg.replace.value
-    if usagetext is not None:
-        # Maybe, if htmltout is True, we should convert
-        # the value to a HTML form...
+
+    if not is_long_form and usagetext:
         return usagetext
-    # Otherwise, look at the pymathics, and builtin docstrings:
+
     builtins = evaluation.definitions.builtin
     pymathics = evaluation.definitions.pymathics
-    bio = pymathics.get(definition.name)
-    if bio is None:
-        bio = builtins.get(definition.name)
+    bio = pymathics.get(definition.name) or builtins.get(definition.name)
 
     if bio is not None:
-        if not is_long_form and hasattr(bio.builtin.__class__, "summary_text"):
-            return bio.builtin.__class__.summary_text
-        from mathics.doc.common_doc import XMLDOC
+        from mathics.doc.doc_entries import DocumentationEntry
 
         docstr = bio.builtin.__class__.__doc__
         title = bio.builtin.__class__.__name__
         if docstr is None:
-            return None
-        usagetext = XMLDOC(docstr, title).text(0)
+            return usagetext
+        docstr = docstr[docstr.find("<dl>") : (docstr.find("</dl>") + 6)]
+        usagetext = DocumentationEntry(docstr, title).text()
         usagetext = re.sub(r"\$([0-9a-zA-Z]*)\$", r"\1", usagetext)
-        return usagetext
-    return None
+    return usagetext
+
+
+def show_definitions(symbol: Symbol, evaluation: Evaluation) -> list:
+    """Return a list of lines describing the definition of `symbol`"""
+    lines = []
+
+    def print_rule(
```

Comment:
`print_rule` is a bad name. There is nothing printed here. We are accumulating into `lines` another `Expression`. Please rename and document.

Reply:
Yep. And probably we want to move this to a `mathics.eval` module.

Reply:
Now it is called `format_rule` that is what it does. If a better name comes up later, we can change it.